### DISCUSSION TITLE
Deploying a java library to maven, getting 422 Unprocessable Entity

### DISCUSSION THREAD

**Discussion by i-make-robots** [community/community#25257] [Actions]
<p>Hello!  I hope you are well today.</p>
<p>I created <a href="https://github.com/Marginallyclever/NodeGraphCore/" class="inline-onebox" rel="noopener nofollow ugc">GitHub - MarginallyClever/NodeGraphCore: Core elements for Data flow programming</a> and I’d like to use it in my other projects as well as offer it to the community.  I’m told the best way is to publish it to GitHub Packages /docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry.  So I installed the first Action offered by Github and bumped the JDK up to 17 ~/blob/4ec681299f20f6768c0e12ceec94ad5ed7937cd4/.github/workflows/maven-publish.yml.  For this I get</p>
<blockquote>
<p>Error:  Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.7:deploy (default-deploy) on project NodeGraphCore: Failed to deploy artifacts: Could not transfer artifact org.marginallyClever:NodeGraphCore:jar:1.0.0 from/to github (/maven.pkg.github.com/MarginallyClever/NodeGraph): transfer failed for /maven.pkg.github.com/MarginallyClever/NodeGraph/org/marginallyClever/NodeGraphCore/1.0.0/NodeGraphCore-1.0.0.jar, status: 422 Unprocessable Entity → [Help 1]</p>
</blockquote>
<p>It is a small thanks that I get the same error if I try to <code>mvn deploy</code> from my local machine - consistent!  I just wish I knew what it meant.  Google hasn’t helped.  Can you please tell me what I’m doing wrong?</p>
<p>My ~/.m2/settings.xml is</p>
<pre><code class="lang-auto">&lt;settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                          https://maven.apache.org/xsd/settings-1.0.0.xsd"&gt;

    &lt;activeProfiles&gt;
        &lt;activeProfile&gt;github&lt;/activeProfile&gt;
    &lt;/activeProfiles&gt;

    &lt;profiles&gt;
        &lt;profile&gt;
            &lt;id&gt;github&lt;/id&gt;
            &lt;repositories&gt;
                &lt;repository&gt;
                    &lt;id&gt;central&lt;/id&gt;
                    &lt;url&gt;https://repo1.maven.org/maven2&lt;/url&gt;
                &lt;/repository&gt;
                &lt;repository&gt;
                    &lt;id&gt;github&lt;/id&gt;
                    &lt;url&gt;https://maven.pkg.github.com/MarginallyClever/NodeGraphCore&lt;/url&gt;
                    &lt;snapshots&gt;
                        &lt;enabled&gt;true&lt;/enabled&gt;
                    &lt;/snapshots&gt;
                &lt;/repository&gt;
            &lt;/repositories&gt;
        &lt;/profile&gt;
    &lt;/profiles&gt;

    &lt;servers&gt;
        &lt;server&gt;
            &lt;id&gt;github&lt;/id&gt;
            &lt;username&gt;i-make-robots&lt;/username&gt;
            &lt;password&gt;*******&lt;/password&gt;
        &lt;/server&gt;
    &lt;/servers&gt;
&lt;/settings&gt;
</code></pre>
<p>My PAT was created following /docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token.  It wasn’t very specific about how much access I should give, which was concerning.</p>
<p>I also tried to <code>mvn deploy</code> locally.  I have no idea how to use the local deployment in my other projects.  Ideally I’d like to confirm library changes don’t break things down-stream before I push out a release.  Tips?</p>
<p>Thank you and stay awesome!</p>
<p>(Sorry for the broken links, github wouldn’t let me write a thorough post because my account is too new. :stuck_out_tongue:)</p>

**Comment by i-make-robots** [comment]
<p>Found my mistake.  In my pom I had somehow written the <code>groupid</code> as ‘org.marginallyClever’ instead of ‘com.marginallyClever’.<br>
Sorry!</p>

**Comment by i-make-robots** [comment]
<p>I’m back, it appears to be broken again.  I don’t get a credentials error, I get <code>422 Unprocessable Entity</code> which tells me very little.  Google doesn’t seem to know what it’s about, either.</p>

**Comment by coliss86** [comment, marked as answer]
<p>I had the same issue, the <code>artifactId</code> and the <code>groupId</code> MUST be written in lower case.<br>
The <code>groupId</code> can be whatever you want :slight_smile:</p>

**Reply by krachynski** [reply]
Is there any explanation as to *why* this is? As far as I can tell, GitHub is the only repository enforcing this rule but no explanation as to why this is.
